src/metabase/lib/query.cljc

```diff
@@ -253,3 +253,43 @@
   [a-query :- ::lib.schema/query
    metric-id :- [:or ::lib.schema.id/legacy-metric :string]]
   (occurs-in-stage-clause? a-query :aggregation #(occurs-in-expression? % :metric metric-id)))
+
+(def ^:private clause-types-order
+  ;; Note that :breakout is never actually sent, but when we get :aggregation we want to drop :breakout too.
```

Comment:
Would using `:summary` be cleaner then? The precedence between `:aggregation` and `:breakout` is somewhat arbitrary (and it's right to left, and reverse lexicographic)).

Reply:
`:summary` would be more transparent, I agree. However, this is effectively a list of keys on a stage, in a specific order. The order of `:aggregation` and `:breakout` is *not* arbitrary.

When previewing some part `:x` we want to remove a **prefix** of the keys in this list, up to but excluding the specified clause `:x`.
- So if you preview `:aggregation` we remove `:limit` and `:order-by`
- If you preview `:filters` we remove `:limit`, `:order-by`, `:aggregation` and `:breakout`.

This is the only order of `:aggregation` and `:breakout` that gives those properties.

I've added an expanded comment to explain that.

Reply:
Isn't the order arbitrary in the sense that I could choose to use `:breakout` instead of `:aggregation` to signify that the summary has to be removed and swap them in the vector? (I'm assuming we can change the callers.)

Reply:
Sure, that's true. The important property is that whichever of `:aggregation` and `:breakout` is sent by the FE when previewing the Summary step is first in this list.